listener/avalon_listener/tcs_work_order_handler.py

```diff
@@ -137,46 +137,49 @@ def WorkOrderGetResult(self, **params):
               as defined in EEA spec 6.1.4
         Returns jrpc response as defined in EEA spec 6.1.2
         """
-        if "workOrderId" in params:
-            wo_id = params["workOrderId"]
-            # Work order status payload should have
-            # data field with work order id as defined in EEA spec section 6.
-            data = {
-                "workOrderId": wo_id
-            }
-            if not is_valid_hex_str(wo_id):
-                logging.error("Invalid work order Id")
-                raise JSONRPCDispatchException(
-                    JsonRpcErrorCode.INVALID_PARAMETER,
-                    "Invalid work order Id",
-                    data
-                )
-
-            # Work order is processed if it is in wo-response table
-            value = self.kv_helper.get("wo-responses", wo_id)
-            if value:
-                response = json.loads(value)
-                if 'result' in response:
-                    return response['result']
-
-                # response without a result should have an error
-                err_code = response["error"]["code"]
-                err_msg = response["error"]["message"]
-
-                if err_code == EnclaveError.ENCLAVE_ERR_VALUE:
-                    err_code = \
-                        WorkOrderStatus.INVALID_PARAMETER_FORMAT_OR_VALUE
-                elif err_code == EnclaveError.ENCLAVE_ERR_UNKNOWN:
-                    err_code = WorkOrderStatus.UNKNOWN_ERROR
-                elif err_code == EnclaveError.ENCLAVE_ERR_INVALID_WORKLOAD:
-                    err_code = WorkOrderStatus.INVALID_WORKLOAD
-                else:
-                    err_code = WorkOrderStatus.FAILED
-                raise JSONRPCDispatchException(
-                    err_code,
-                    err_msg,
-                    data
-                )
+
+        input_json_str = params["raw"]
+        input_value_json = json.loads(input_json_str)
+        req_validator = WorkOrderRequestValidator()
+        valid, err_msg = req_validator.workorder_get_result_validation(
+            input_value_json["params"])
+        if not valid:
+            raise JSONRPCDispatchException(
+                WorkOrderStatus.INVALID_PARAMETER_FORMAT_OR_VALUE,
+                err_msg
+            )
+
+        # Work order status payload should have
+        # data field with work order id as defined in EEA spec section 6.
+        data = {
```

Comment:
This could be moved to line num #144 maybe

Reply:
workorder ID  cannot be fetched before validation. Hence cannot move before validation